godel_process_path_generation/src/mesh_importer.cpp

```diff
@@ -261,28 +261,56 @@ bool MeshImporter::calculateBoundaryData(const pcl::PolygonMesh &input_mesh)
   return true;
 }
 
-void MeshImporter::computeLocalPlaneFrame(const Eigen::Hyperplane<double, 3> &plane, const Vector3d &centroid)
+void MeshImporter::computeLocalPlaneFrame(const Eigen::Hyperplane<double, 3> &plane, const Vector4d &centroid, const Cloud& cloud)
 {
-  Eigen::Vector3d origin = plane.projection(centroid);       // Project centroid onto plane
-
-  // Check if z_axis (plane normal) is closely aligned with world x_axis:
-  // If not, construct transform rotation from X,Z axes. Otherwise, use Y,Z axes.
+  Eigen::Vector3d origin = plane.projection(centroid.head(3));       // Project centroid onto plane
```

Comment:
You should use `centroid.head<3>()` rather than `centroid.head(3)`
See [Eigen: Block operations](http://eigen.tuxfamily.org/dox/group__TutorialBlockOperations.html)